### PR TITLE
[CONTRIB] New MulticolumnValuesProductEqualToSingleColumn Expectation

### PR DESCRIPTION
**Pull request by calvingdu**
Adding an expectation very similar to expect_multicolumn_sum_values_to_be_equal_to_single_column but using multiplication

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: contrib/experimental/great_expectations_experimental/expectations/expect_multicolumn_product_values_to_be_equal_to_single_column.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        #     raise InvalidExpectationConfigurationError(str(e))
+
+    # This object contains metadata for display in the public Gallery
+    # <snippet>
+    library_metadata = {
+        "tags": [
+            "multi-column expectation",
+            "multi-column product values to be equal to single column",
+        ],
+        "contributors": ["@AsaFLachisch", "@mkopec87"],
```
Comment (anthonyburdi) [inline contrib/experimental/great_expectations_experimental/expectations/expect_multicolumn_product_values_to_be_equal_to_single_column.py]: 😄 
```suggestion
        "contributors": ["@calvingdu"],
```

**Review by anthonyburdi — COMMENTED** [review]
Thanks for this contribution! I just have one small suggestion to use your name as the contributor.
Also if you can sign the CLA with the email address used to sign your commits then I'll be able to approve and merge:
Users missing a CLA: [calvindu@dhcp-206-12-66-221.ubcsecure.wireless.ubc.ca](mailto:calvindu@dhcp-206-12-66-221.ubcsecure.wireless.ubc.ca)

**Comment by calvingdu** [comment]
> Thanks for this contribution! I just have one small suggestion to use your name as the contributor. Also if you can sign the CLA with the email address used to sign your commits then I'll be able to approve and merge: Users missing a CLA: [calvindu@dhcp-206-12-66-221.ubcsecure.wireless.ubc.ca](mailto:calvindu@dhcp-206-12-66-221.ubcsecure.wireless.ubc.ca)

Hi, I'm not sure why my CLA isn't working considering I was able to push changes before. I'm still under the same email/github account and tried refilling the CLA form

**Comment by calvingdu** [comment]
@cla-bot check

**Review by anthonyburdi — APPROVED** [review]
Thank you! LGTM!

**Pull request merged**